pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyService.java

```diff
@@ -356,6 +374,10 @@ public void close() throws IOException {
         }
     }
 
+    private synchronized void resetMetricsServlet() {
```

Comment:
getMetricsServlet should be `synchronized` as well (spotbugs should not pass)

Reply:
done